### PR TITLE
Don't clobber `in` and `name` fields in body/form parameters

### LINKED ISSUE #2: `in` and `name` fields for body parameters get clobbered
When defining a path that takes a body parameter, the first request to the path clobbers the `in` and `name` fields for the body parameter, resulting in these warnings upon subsequent requests:

```
Use of uninitialized value $in in string eq at /home/zakame/src/swagger2/lib/Mojolicious/Plugin/Swagger2.pm line 263, <DATA> line 47.
Use of uninitialized value $in in string eq at /home/zakame/src/swagger2/lib/Mojolicious/Plugin/Swagger2.pm line 263, <DATA> line 47.
Use of uninitialized value $in in string eq at /home/zakame/src/swagger2/lib/Mojolicious/Plugin/Swagger2.pm line 263, <DATA> line 47.
Use of uninitialized value $name in anonymous hash ({}) at /home/zakame/src/swagger2/lib/Mojolicious/Plugin/Swagger2.pm line 272, <DATA> line 47.
Use of uninitialized value $name in anonymous hash ({}) at /home/zakame/src/swagger2/lib/Mojolicious/Plugin/Swagger2.pm line 272, <DATA> line 47.
Use of uninitialized value $name in hash element at /home/zakame/src/swagger2/lib/Mojolicious/Plugin/Swagger2.pm line 275, <DATA> line 47.
```

Checking on the code, it seems that the plugin validates input for each parameter, but clobbers the parameter settings with `$p->{schema}` if the schema field is detected.  Additionally, when finding the `in` field to be either a body or form parameter, it defaults the `$value` to `$body->{name}` when it may not be necessarily the case, per [spec][0]:

[0]: https://github.com/swagger-api/swagger-spec/blob/master/versions/2.0.md#parameter-object-

>Body - The payload that's appended to the HTTP request. Since there can only be one payload, there can only be one body parameter. The name of the body parameter has no effect on the parameter itself and is used for documentation purposes only. Since Form parameters are also in the payload, body and form parameters cannot exist together for the same operation.

FWIW, I am writing an API where I do not necessarily use the `name` field of a parameter in a JSON body (preferring to directly specify the fields for the object I want to create instead.)

### PR DESCRIPTION
**Pull request by zakame**
Fixes #2.  I'll try to follow this up with a corresponding test script.

### CODE REVIEW

**Comment by jhthorsen** [comment]
I merged everything, except the check for complete $body - Which I like, but not sure if is according to spec.

I will make a new release to CPAN tomorrow.

File: lib/Mojolicious/Plugin/Swagger2.pm [hunk 1]
```diff
@@ -264,7 +265,7 @@ sub _validate_input {
       = $in eq 'query'  ? $query->param($name)
       : $in eq 'path'   ? $c->stash($name)
       : $in eq 'header' ? $headers->header($name)
-      :                   $body->{$name};
+      :                   $body->{$name} || $body;
```
Comment (jhthorsen) [inline lib/Mojolicious/Plugin/Swagger2.pm]: Feel free to re-open if you can point to the spec where it says the schema can be tested against a complete json $body.

**Comment by jhthorsen** [comment]
Thank you for your contribution!

**Comment by jhthorsen** [comment]
@zakame: I found this in https://github.com/swagger-api/swagger-spec/blob/master/versions/2.0.md#parameterObject

   The name of the body parameter has no effect on the parameter itself and is used for documentation purposes only

So this means that you are right all along! And luckily it makes perfect sense :+1:

Making sure the implementation match the documentation now.

**Comment by zakame** [comment]
Great to see it! :+1:

**Pull request closed without merging**